apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/Blockchain.java

```diff
@@ -64,6 +65,8 @@
 
     void commit(Block block);
 
+    Long getBlockCount(TransactionalDataSource dataSource, int from, int to);
```

Comment:
I think, that blockchain should not take datasource as parameter. Maybe specify datasource by id?

Reply:
We don't know the DataSource by Id beforehand, only in test it's 'kind of fixed'.
Here I wanted decrease injected dependencies in unit test for checking rows in main or shard db.
I can remove those methods on Blockchain service, but Inject only BlockDao + TransactionDao into unit test.